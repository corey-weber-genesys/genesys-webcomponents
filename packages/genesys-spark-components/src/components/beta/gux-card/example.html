<h1>gux-card-beta</h1>

<div class="container">
  <gux-card-beta accent="outline">
    <h2>Outline</h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam convallis
      erat fringilla tristique condimentums.
    </p>
  </gux-card-beta>

  <gux-card-beta accent="bordered">
    <h2>Bordered</h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam convallis
      erat fringilla tristique condimentums.
    </p>
  </gux-card-beta>

  <gux-card-beta accent="filled">
    <h2>Filled</h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam convallis
      erat fringilla tristique condimentums.
    </p>
  </gux-card-beta>

  <gux-card-beta accent="raised">
    <h2>Raised</h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam convallis
      erat fringilla tristique condimentums.
    </p>
  </gux-card-beta>

  <gux-card-beta accent="borderless">
    <h2>Borderless</h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam convallis
      erat fringilla tristique condimentums.
    </p>
  </gux-card-beta>
</div>

<style>
  .container {
    display: flex;
    flex-wrap: wrap;
    gap: 20px;
    padding: 20px;
    background-color: #f2f2f2;
  }

  gux-card-beta {
    width: 200px;
  }
</style>
