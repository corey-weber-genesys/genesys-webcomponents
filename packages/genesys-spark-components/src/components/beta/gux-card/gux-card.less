@import (reference) '../../../style/color.less';
@import (reference) '../../../style/mixins.less';
@import (reference) '../../../style/spacing.less';
@import (reference) '../../../style/typography.less';
@import (reference) '../../../style/shadows.less';

:host {
  display: block;
  width: fit-content;
}

.gux-card {
  box-sizing: border-box;
  padding: @gux-spacing-medium;
  background-color: @gux-grey-100;
  border: 1px solid @gux-grey-60;
  border-radius: @gux-spacing-2xs;

  &.gux-outline,
  &.gux-bordered {
    background-color: @gux-grey-100;
  }

  &.gux-filled {
    background-color: @gux-grey-90;
  }

  &.gux-raised {
    .gux-shadow-10();
  }

  &.gux-borderless {
    border-color: transparent;
  }
}
