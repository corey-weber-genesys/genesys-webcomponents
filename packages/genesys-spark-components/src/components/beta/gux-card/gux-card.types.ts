export type GuxCardAccent = 'outline' | 'filled' | 'raised';

export type GuxCardAccentV4 = 'bordered' | 'filled' | 'raised' | 'borderless';
