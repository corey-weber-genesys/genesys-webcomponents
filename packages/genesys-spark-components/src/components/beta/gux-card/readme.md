# gux-card-beta



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description  | Type                                                              | Default     |
| -------- | --------- | ------------ | ----------------------------------------------------------------- | ----------- |
| `accent` | `accent`  | Card Accent. | `"bordered" \| "borderless" \| "filled" \| "outline" \| "raised"` | `'outline'` |


## Slots

| Slot | Description      |
| ---- | ---------------- |
|      | Content of card. |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
