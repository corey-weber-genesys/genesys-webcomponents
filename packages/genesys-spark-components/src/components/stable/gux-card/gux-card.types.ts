export type GuxCardAccent = 'bordered' | 'raised' | 'borderless';
