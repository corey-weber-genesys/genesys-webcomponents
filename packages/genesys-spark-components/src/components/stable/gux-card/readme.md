# gux-card



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description  | Type                                     | Default      |
| -------- | --------- | ------------ | ---------------------------------------- | ------------ |
| `accent` | `accent`  | Card Accent. | `"bordered" \| "borderless" \| "raised"` | `'bordered'` |


## Slots

| Slot | Description      |
| ---- | ---------------- |
|      | Content of card. |


## CSS Custom Properties

| Name                                    | Description |
| --------------------------------------- | ----------- |
| `--gse-ui-card-backgroundColor`         |             |
| `--gse-ui-card-borderless-border-color` |             |
| `--gse-ui-card-borderless-border-style` |             |
| `--gse-ui-card-borderless-border-width` |             |
| `--gse-ui-card-borderRadius`            |             |
| `--gse-ui-card-default-border-color`    |             |
| `--gse-ui-card-default-border-style`    |             |
| `--gse-ui-card-default-border-width`    |             |
| `--gse-ui-card-padding`                 |             |
| `--gse-ui-card-raised-border-color`     |             |
| `--gse-ui-card-raised-border-style`     |             |
| `--gse-ui-card-raised-border-width`     |             |
| `--gse-ui-card-raised-boxShadow-blur`   |             |
| `--gse-ui-card-raised-boxShadow-color`  |             |
| `--gse-ui-card-raised-boxShadow-spread` |             |
| `--gse-ui-card-raised-boxShadow-x`      |             |
| `--gse-ui-card-raised-boxShadow-y`      |             |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
