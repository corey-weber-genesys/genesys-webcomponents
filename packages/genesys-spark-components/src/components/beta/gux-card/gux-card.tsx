import { Component, h, JSX, Prop, Element } from '@stencil/core';
import { trackComponent } from '@utils/tracking/usage';

import { GuxCardAccent, GuxCardAccentV4 } from './gux-card.types';

/**
 * @slot - Content of card.
 */

@Component({
  styleUrl: 'gux-card.scss',
  tag: 'gux-card-beta',
  shadow: true
})
export class GuxCard {
  /*
   * Reference to the host element.
   */
  @Element()
  root: HTMLElement;

  /**
   * Card Accent.
   */
  @Prop()
  accent: GuxCardAccent | GuxCardAccentV4 = 'outline';

  componentWillLoad(): void {
    trackComponent(this.root, { variant: this.accent });
  }

  render(): JSX.Element {
    return (
      <div
        class={{
          'gux-card': true,
          [`gux-${this.accent}`]: true
        }}
      >
        <slot />
      </div>
    ) as JSX.Element;
  }
}
