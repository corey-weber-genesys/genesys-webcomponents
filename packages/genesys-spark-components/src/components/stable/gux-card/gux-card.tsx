import { Component, h, JSX, Prop, Element } from '@stencil/core';
import { trackComponent } from '@utils/tracking/usage';

import { GuxCardAccent } from './gux-card.types';

/**
 * @slot - Content of card.
 */

@Component({
  styleUrl: 'gux-card.scss',
  tag: 'gux-card',
  shadow: true
})
export class GuxCard {
  /*
   * Reference to the host element.
   */
  @Element()
  root: HTMLElement;

  /**
   * Card Accent.
   */
  @Prop()
  accent: GuxCardAccent = 'bordered';

  componentWillLoad(): void {
    trackComponent(this.root, { variant: this.accent });
  }

  render(): JSX.Element {
    return (
      <div
        class={{
          'gux-card': true,
          [`gux-${this.accent}`]: true
        }}
      >
        <slot />
      </div>
    ) as JSX.Element;
  }
}
