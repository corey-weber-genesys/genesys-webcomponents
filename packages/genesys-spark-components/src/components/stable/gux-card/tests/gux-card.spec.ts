import { newSpecPage } from '@stencil/core/testing';
import { GuxCard } from '../gux-card';

const components = [GuxCard];
const language = 'en';

describe('#render different card accents', () => {
  [
    '<gux-card><span>This is a content slot</span></gux-card>',
    '<gux-card accent="bordered"><span>This is a content slot</span></gux-card>',
    '<gux-card accent="raised"><span>This is a content slot</span></gux-card>',
    '<gux-card accent="borderless"><span>This is a content slot</span></gux-card>'
  ].forEach((html, index) => {
    it(`should render component as expected (${index + 1})`, async () => {
      const page = await newSpecPage({ components, html, language });

      expect(page.root).toMatchSnapshot();
    });
  });
});
