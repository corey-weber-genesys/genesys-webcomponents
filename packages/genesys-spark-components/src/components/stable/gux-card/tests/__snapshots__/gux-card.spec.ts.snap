// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render different card accents should render component as expected (1) 1`] = `
<gux-card>
  <mock:shadow-root>
    <div class="gux-bordered gux-card">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <span>
    This is a content slot
  </span>
</gux-card>
`;

exports[`#render different card accents should render component as expected (2) 1`] = `
<gux-card accent="bordered">
  <mock:shadow-root>
    <div class="gux-bordered gux-card">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <span>
    This is a content slot
  </span>
</gux-card>
`;

exports[`#render different card accents should render component as expected (3) 1`] = `
<gux-card accent="raised">
  <mock:shadow-root>
    <div class="gux-card gux-raised">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <span>
    This is a content slot
  </span>
</gux-card>
`;

exports[`#render different card accents should render component as expected (4) 1`] = `
<gux-card accent="borderless">
  <mock:shadow-root>
    <div class="gux-borderless gux-card">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <span>
    This is a content slot
  </span>
</gux-card>
`;
